doc/source/whatsnew/v1.1.0.rst

```diff
@@ -352,6 +352,8 @@ Other
   instead of ``TypeError: Can only append a Series if ignore_index=True or if the Series has a name`` (:issue:`30871`)
 - Set operations on an object-dtype :class:`Index` now always return object-dtype results (:issue:`31401`)
 - Bug in :meth:`AbstractHolidayCalendar.holidays` when no rules were defined (:issue:`31415`)
+- :meth:`DataFrame.T` now has type annotations in line with :meth:`DataFrame.transpose`. Annotations also extended to :class:`Categorical`
```

Comment:
You can remove these; we haven’t added whatsnew notes for annotations

Reply:
Done